TestFoundation/TestNSArray.swift

```diff
@@ -53,6 +56,45 @@ class TestNSArray : XCTestCase {
         XCTAssertEqual(array.count, 0)
         XCTAssertEqual(array2.count, 2)
     }
+
+    func test_constructors() {
+        let arrayNil = NSArray(objects: nil, count: 0)
+        XCTAssertEqual(arrayNil.count, 0)
+
+        let array1 = NSArray(object: "foo")
+        XCTAssertEqual(array1.count, 1)
+
+        let testStrings: [AnyObject] = [NSString(string:"foo"), NSString(string: "bar")]
+        testStrings.withUnsafeBufferPointer { ptr in
+            let array2 = NSArray(objects: ptr.baseAddress, count: 1)
+            XCTAssertEqual(array1, array2)
+        }
+
+        let array3 = NSArray(objects: "foo" as NSString, "bar" as NSString, "baz" as NSString)
+        XCTAssertEqual(array3.count, 3)
+        let array4 = NSArray(array: ["foo", "bar", "baz"])
+        XCTAssertEqual(array4.count, 3)
+        let array5 = NSArray(arrayLiteral: "foo", "bar", "baz")
+        XCTAssertEqual(array5.count, 3)
+        XCTAssertEqual(array3, array4)
+        XCTAssertEqual(array3, array5)
+        XCTAssertEqual(array4, array5)
+
+    }
+
+    func test_constructorWithCopyItems() {
+        let foo = "foo" as NSMutableString
+
+        let array1 = NSArray(array: [foo], copyItems: false)
+        let array2 = NSArray(array: [foo], copyItems: true)
+
+        XCTAssertEqual(array1[0] as! String, "foo")
+        XCTAssertEqual(array2[0] as! String, "foo")
+
+        foo.append("1")
+        XCTAssertEqual(array1[0] as! String, "foo1")
+        XCTAssertEqual(array2[0] as! String, "foo")
```

Comment:
@phausler Is this test (which fails on CI) testing the `copyItems` correctly? The test passed against Darwin Foundation but Im not sure if the issue is the test or the `init(array:copyItems:)` constructor.